CenterForOpenScience/osf.io
# Feature/APIv2 Create Registration Field Changes [ENG-1087]

⭐️ Introduces new API version, make sure this doesn't need to get bumped before merging 

## Purpose

Cleaner field names and to avoid name collisions in ember (children) when creating a registration with v2 (POST /v2/registrations/)

## Changes

On RegistrationCreateSerializer, update the field names passed in when creating a registration. This is needed for parity, and has been pulled out of the DraftNode work.

- rename `children` to `included_node_ids` (list field of guid strings)
- remove registration_choice and lift_embargo – instead, allow setting `embargo_end_date` on creation (to make public immediately, use null)
- rename draft_registration to `draft_registration_id` or something
- New API version introduced - 2.18 - but will probably have to be bumped before merge 
- 2.18 and later expect the new fields when creating a registration.
- Pre 2.18, use the same fields as before to create a registration. 

## QA Notes
- Test creating registrations in the API with old and new versions.
- Use old payload for old versions and new payload below for new versions
- Verify legacy creating registration still works (as legacy create uses APIv2)
- POST v2/registrations/?version=2.18, new expected request:
```json
{
    "data": {
        "type": "registrations",
        "attributes": {
            "draft_registration_id": "<draft_registration_id>",
            "embargo_end_date": "2020-01-29T12:00:00",
             "included_node_ids": ["child_id", "grandchild_id"]
        }
    }
 }
```
- If no `embargo_end_date`, registration is immediate.
- `draft_registration_id` is a required field 
- `included_node_ids` instead of `children`
- Same registration rules still apply: draft can't have been registered before, embargo end date must not be more than 4 years in the future, etc.
## Documentation

Will need API documentation if we decide to go this route

## Side Effects



## Ticket

https://openscience.atlassian.net/browse/ENG-1087
